.bash_profile.khan

```diff
@@ -53,19 +53,22 @@ fi
 # Add a mykeeper alias to run keeper with KA config
 alias mykeeper="keeper --config $HOME/.keeper-config.json"
 
-# Setting this allows us to store ssh-keys in the keychain without generating
-# a warning.  See ssh-add man page.
-export APPLE_SSH_ADD_BEHAVIOR=macos
+if [ "$(uname -s)" = "Darwin" ]; then
+    # Setting this allows us to store ssh-keys in the keychain without generating
+    # a warning.  See ssh-add man page.
+    export APPLE_SSH_ADD_BEHAVIOR=macos
+
+    # Add ssh keys stored in the keychain to the ssh-agent
+    # Note: IF you have an identity in ~/.ssh and DO NOT have a passphrase already
+    #   in the keychain, you will be prompted for a passphrase.  This is because
+    #   ssh-add will try to add the key to the agent, and will prompt for a
+    #   passphrase if it doesn't have one in the keychain.
+    #   If you have a passphrase in the keychain, you will not be prompted.
+    #   If you don't have an identity in ~/.ssh, you will not be prompted.
+    #   If you have an identity in ~/.ssh and DO have a passphrase in the keychain,
+    #   you will not be prompted.
+    #   (Being prompted should NOT happen if you ran ssh-add -K when you first
+    #    created the key.)
+    ssh-add -K
+fi
```

Comment:
Do we want to do just an `ssh-add` in the non-osx case?  Or is that redundant?

Reply:
For anyone with a passphrase on their SSH key, it will block the shell to prompt for the passphrase. I'd rather be prompted the first time it's used, rather than immediately the first time I open a terminal, so my preference would be to not do it.

Reply:
OK by me!  People on linux might not be running ssh-agent anyway.